spec/ktl/reassigner_spec.rb

```diff
@@ -280,6 +280,60 @@ module Ktl
           expect(reassigned.size).to eq(20)
         end
       end
+
+      context 'with a current assignment' do
+        let :reassignment do
+          r = Scala::Collection::Map.empty
+          10.times.map do |partition|
+            topic_partition = Kafka::TopicAndPartition.new('topic1', partition)
+            replicas = scala_int_list([0, 2])
+            r += Scala::Tuple.new(topic_partition, replicas)
+          end
+          r
+        end
+
+        let :current_assignment do
+          r = Scala::Collection::Map.empty
+          10.times.map do |partition|
+            topic_partition = Kafka::TopicAndPartition.new('topic1', partition)
+            replicas = scala_int_list([0, 1])
+            r += Scala::Tuple.new(topic_partition, replicas)
+          end
+          r
+        end
+
+        let :duplicated_reassignment do
+          r = Scala::Collection::Map.empty
+          10.times.map do |partition|
+            topic_partition = Kafka::TopicAndPartition.new('topic1', partition)
+            replicas = scala_int_list([0, 1, 2])
+            r += Scala::Tuple.new(topic_partition, replicas)
+          end
+          r
+        end
+
+        before do
+          allow(zk_client).to receive(:replica_assignment_for_topics).and_return(current_assignment)
+        end
+
+        let :final_json do
+          zk_utils.format_as_reassignment_json(reassignment)
+        end
+
+        let :duplicated_json do
+          zk_utils.format_as_reassignment_json(duplicated_reassignment)
+        end
+
+        it 'creates a reassignment that duplicates partitions that are being reassigned' do
+          reassigner.execute(reassignment)
+          expect(zk_client).to have_received(:reassign_partitions).with(duplicated_json)
```

Comment:
Is there any chance that the thing that produces the JSON could produce JSON that is not always identical with the JSON produced by this test? It looks brittle to me. A reordering of object properties would be an equivalent JSON, but fail this test, for example.

How about capturing the call and asserting on the objects instead:

```ruby
something = nil
allow(zk_client).to receive(:reassign_partitions) do |json|
  something = JSON.load(json)
end
reassigner.execute(reassignment)
expect(something).to eq(some_other_thing)
```

(I'm not sure about the name, so I used `something`, but that is obviously a terrible name)